amd_openvx_extensions/amd_rpp/source/tensor/PythonFunction.cpp

```diff
@@ -0,0 +1,310 @@
+/*
+Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+
+// Local copy of rocAL Python bridge C ABI (keep in sync with rocAL)
+#ifndef ROCAL_PY_MAX_TENSOR_DIMS
+#define ROCAL_PY_MAX_TENSOR_DIMS 5
+#endif
+
+typedef struct {
+    size_t num_dims;                          /* e.g., 4 for [N,H,W,C] */
+    size_t shape[ROCAL_PY_MAX_TENSOR_DIMS];   /* lengths per dimension */
+    size_t strides[ROCAL_PY_MAX_TENSOR_DIMS]; /* strides in elements */
```

Comment:
does rocAL take strides in elements? I thought we changed that

Reply:
In this code it assumes that ROCAL_PY_MAX_TENSOR_DIMS is same as RPP_MAX_TENSOR_DIMS. Then why are you adding a new enum. Please use RPP_MAX_TENSOR_DIMS instead